src/Models/ArduinoDeviceBase.ts

```diff
@@ -166,17 +165,15 @@ export abstract class ArduinoDeviceBase implements Device {
 
   abstract get version(): string;
 
-  async generateCppPropertiesFile(
-      type: ScaffoldType, board: Board,
-      fileInfo: TemplateFileInfo): Promise<void> {
-    const targetFolder = path.join(this.deviceFolder, fileInfo.targetPath);
-    if (!await FileUtility.directoryExists(type, targetFolder)) {
-      await FileUtility.mkdirRecursively(type, targetFolder);
+  async generateCppPropertiesFile(type: ScaffoldType, board: Board):
+      Promise<void> {
+    if (!await FileUtility.directoryExists(type, this.vscodeFolderPath)) {

```

Comment:
can we merge them to ensureDirectory()?

Reply:
Yes. Consider fixing this in next iteration.

Reply:
Yes. Consider fixing this in next iteration.